### ISSUE TITLE
Allow per site language selection

### ISSUE THREAD

**Issue by jidanni** [hlawuleka/page-language-switcher-webextension#3]
Pleas add a way for sites/pages matching
a.b.c/xxx?yyy&zzz
to use language P, instead of language Q.

You see some sites work better with some locale settings, and others work better with others,
and we don't want to manually change them each time. Nor do we want to bother contacting the owner and asking him to fix his site.

**Comment by jidanni** [comment]
I cannot tell if you are working on this as all I see is
```refactor: add functionality yo store keys in local storage and make s…```
and nowhere can I find what that s work was.

**Comment by jidanni** [comment]
Also s/yo/to/.

**Comment by hlawuleka** [comment]
I am not sure I understand you entirely. Are you saying I should add something along the lines:

If you are at www.abc.com load a specific set of languages that the site supports?

**Comment by hlawuleka** [comment]
> I cannot tell if you are working on this as all I see is
> `refactor: add functionality yo store keys in local storage and make s…`
> and nowhere can I find what that s work was.

**Comment by jidanni** [comment]
Yes:
Here's the deal. The rotten
https://maps.nlsc.gov.tw/
needs to have the zh-TW moved in front on
```Accept-Language: en-US,en;q=0.9,zh-TW;q=0.8,zh;q=0.7```
otherwise is is a big mess:
![0](https://user-images.githubusercontent.com/1230959/45429199-6266e580-b6d5-11e8-81fd-62e77978525e.jpg)
However I only want to do that for https://maps.nlsc.gov.tw/ !

**Comment by jidanni** [comment]
And no, I don't want to use the "```English```" version of  https://maps.nlsc.gov.tw/ .

**Comment by jidanni** [comment]
Yes I told the site owners it was all their fault, but they are ninnies.

**Comment by jidanni** [comment]
Anyway, what I want is real simple:
If site URL matches pattern A then send Accept-Language header X.
else If site URL matches pattern B then send Accept-Language header Y...
Else send the default Accept-Language header.

**Comment by hlawuleka** [comment]
I think I understand you now. I have a question regarding "If site URL matches pattern A then send Accept-Language header X." How would the extension gather that you are translating on site A, what I mean is, where would that logic sit? Because that's very specific to one need and I am not sure it'd be ideal to implement that. Unless you wanted something/an option where you could have an input field where you can define that yourself - but the problem with this is that users will not know and shouldn't have to know the working(s) of site A as far as modifying headers is concerned. So I don't know how we could go about this. Any implementation suggestions?

**Comment by jidanni** [comment]
That way the user could pleasantly browse all day, never having to touch anything to "shift gears", as the particular Accept-Language header needed to make each broken site function properly would automatically be sent for him.

Yes, some sites make certain assumptions and the only way to fulfill them is have the right  Accept-Language header they need to operate properly sent to them.

**Comment by hlawuleka** [comment]
Please see my comment and question above, please.

**Comment by jidanni** [comment]
> I think I understand you now. I have a question regarding "If site URL matches pattern A then send Accept-Language header X." How would the extension gather that you are translating on site A

Assume the same user likes to use sites C, D, E in Chinese, and G, H, I
in English, despite all six of them offering both languages.

OK, I assume the way the extension currently works is one needs to punch
a button six times, if browsing C G D H E I.

I'm just wanting a way for him to not need to punch the button at all,
because the extension will remember it for him.

There could be a list
DEFAULT: English
/URL PATTERN C/ {Chinese}
/URL PATTERN D/ {Chinese}
/URL PATTERN E/ {Chinese}

And the extension would say "oh, I see he is connecting to C, well that
matches Chinese. So I'll attach the Chinese Accept-Language header".
